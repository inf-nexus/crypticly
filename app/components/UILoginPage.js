// @flow

import React, { PureComponent } from 'react';

class UILoginPage extends PureComponent {
  render() {
    return <h1>hello world!!!</h1>;
  }
}

export default UILoginPage;
