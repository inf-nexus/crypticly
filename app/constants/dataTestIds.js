export const LOGIN_BUTTON_ID = 'login-button-id';
export const LOGIN_USERNAME_INPUT_ID = 'login-username-input-id';
export const LOGIN_PASSWORD_INPUT_ID = 'login-password-input-id';
