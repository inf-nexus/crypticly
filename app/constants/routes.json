{
  "LOGIN": "/login",
  "HOME": "/home",
  "PASSWORD": "/home/password",
  "PASSWORD_FORM":"/home/password/form",
  "PASSWORD_GENERATOR": "/home/password/generator"
}
