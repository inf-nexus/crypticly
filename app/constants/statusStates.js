export const UNINITIALIZED = 'UNINITIALIZED';
export const PENDING = 'PENDING';
export const SUCCESS = 'SUCCESS';
export const FAILED = 'FAILED';
