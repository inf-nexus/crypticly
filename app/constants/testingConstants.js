import Password from 'constants/records/Password';

const password1 = {
  username: 'foo',
  password: 'bar',
  title: 'Hello World',
  url: 'https://google.com',
  group: 'Social'
};

const password2 = {
  username: 'foo',
  password: 'bar',
  title: 'Singularity',
  url: 'https://google.com',
  group: 'Social'
};

const password3 = {
  username: 'foo',
  password: 'bar',
  title: 'Best Bank',
  url: 'https://google.com',
  group: 'Banking'
};

export const passwordGroups = [
  { group: 'Social', passwords: [password1, password2] },
  { group: 'Banking', passwords: [password3] }
];

export const credentials = {
  username: 'foo',
  password: 'bar'
};
