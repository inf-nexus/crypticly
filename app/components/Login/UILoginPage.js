// @flow

import React, { PureComponent } from 'react';
import { Link, Redirect } from 'react-router-dom';
import styled from 'styled-components';
import { partial } from 'lodash';

import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

import UIFormInput from 'ui-components/UIFormInput';
import UIButton from 'ui-components/UIButton';

import routes from 'constants/routes';
import * as dataTestIds from 'constants/dataTestIds';

const TITLE = 'Crypticly';
const USERNAME = 'username';
const PASSWORD = 'password';

const UILoginPageContainer = styled.div`
  justify-content: center;
  align-items: center;
  margin-top: 15vh;
`;

const LoginContainer = styled(Paper)`
  margin: auto;
  width: 500px;
  height: 300px;
  padding: 30px;
  display: flex;
  flex-direction: column;
`;

const LoginFormContainer = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  flex: 3;
`;

const Header = styled(Typography)`
  flex: 1;
`;

type Props = {
  onLoginClicked: (username: string, password: string) => void,
  authenticated: boolean
};

type State = {
  username: string,
  password: string
};

class LoginForm extends PureComponent<Props, State> {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      password: ''
    };

    this.handleInputUpdate = this.handleInputUpdate.bind(this);
  }

  handleInputUpdate(field, event) {
    this.setState({ [field]: event.target.value });
  }

  render() {
    const { username, password } = this.state;
    const { onLoginClicked, authenticated } = this.props;

    let content;

    if (authenticated) {
      content = <Redirect to={routes.HOME} />;
    } else {
      content = (
        <UILoginPageContainer>
          <LoginContainer>
            <Header align="center" variant="h3">
              {TITLE}
            </Header>
            <LoginFormContainer>
              <UIFormInput
                label={USERNAME}
                value={username}
                onChange={partial(this.handleInputUpdate, 'username')}
                data-test-id={dataTestIds.LOGIN_USERNAME_INPUT_ID}
              />
              <UIFormInput
                label={PASSWORD}
                value={password}
                onChange={partial(this.handleInputUpdate, 'password')}
                type="password"
                data-test-id={dataTestIds.LOGIN_PASSWORD_INPUT_ID}
              />
              <UIButton
                title="Submit"
                onClick={partial(onLoginClicked, username, password)}
                data-test-id={dataTestIds.LOGIN_BUTTON_ID}
              />
            </LoginFormContainer>
          </LoginContainer>
        </UILoginPageContainer>
      );
    }

    return content;
  }
}

export default LoginForm;
