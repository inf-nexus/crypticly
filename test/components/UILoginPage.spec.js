import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UILoginPage from '../../app/components/Login/UILoginPage';
import * as dataTestIds from 'constants/dataTestIds';

import { findComponentByDataTestId } from '../testUtils';

Enzyme.configure({ adapter: new Adapter() });

function setup(componentState = {}) {
  const props = {
    onLoginClicked: spy(),
    authenticated: false
  };

  const component = shallow(<UILoginPage {...props} />);
  component.setState(componentState);
  const loginButton = findComponentByDataTestId(
    component,
    dataTestIds.LOGIN_BUTTON_ID
  );
  const usernameInput = findComponentByDataTestId(
    component,
    dataTestIds.LOGIN_USERNAME_INPUT_ID
  );
  const passwordInput = findComponentByDataTestId(
    component,
    dataTestIds.LOGIN_PASSWORD_INPUT_ID
  );

  return {
    component,
    loginButton,
    usernameInput,
    passwordInput,
    props
  };
}

describe('UILoginPage component', () => {
  it('renders correctly', () => {
    const { props } = setup();
    const tree = renderer.create(<UILoginPage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('login button should call onLoginClicked with correct paramaters', () => {
    const componentState = {
      username: 'hello',
      password: 'world'
    };
    const { component, loginButton, props } = setup(componentState);
    loginButton.simulate('click');
    expect(
      props.onLoginClicked.calledWith(
        componentState.username,
        componentState.password
      )
    ).toBe(true);
  });

  it('should update state when handleInputUpdate is called', () => {
    const { component, usernameInput, passwordInput } = setup();
    const USERNAME = 'hello';
    const PASSWORD = 'world';

    usernameInput.simulate('change', { target: { value: USERNAME } });
    passwordInput.simulate('change', { target: { value: PASSWORD } });

    expect(component.state().username).toBe(USERNAME);
    expect(component.state().password).toBe(PASSWORD);
  });
});
