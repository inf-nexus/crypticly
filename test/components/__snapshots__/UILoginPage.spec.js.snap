// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UILoginPage component renders correctly 1`] = `
<div
  className="sc-bxivhb dJlUyI"
>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-ifAKCX jjcFGL"
  >
    <h3
      className="MuiTypography-root-28 MuiTypography-h3-42 MuiTypography-alignCenter-51 sc-bZQynM jQAXPO"
    >
      Crypticly
    </h3>
    <form
      className="sc-EHOje hdQlJB"
    >
      <div
        className="sc-bdVaJa jpqvBK"
      >
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37 sc-htpNat daKUEJ"
        >
          username
          :
        </p>
        <div
          className="MuiInputBase-root-77 MuiInput-root-64 sc-bwzfXH fWFEyb"
          onClick={[Function]}
          required={true}
        >
          <input
            autoComplete="false"
            className="MuiInputBase-input-87 MuiInput-input-72"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="sc-bdVaJa jpqvBK"
      >
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37 sc-htpNat daKUEJ"
        >
          password
          :
        </p>
        <div
          className="MuiInputBase-root-77 MuiInput-root-64 sc-bwzfXH fWFEyb"
          onClick={[Function]}
          required={true}
        >
          <input
            autoComplete="false"
            className="MuiInputBase-input-87 MuiInput-input-72 MuiInputBase-inputType-90 MuiInput-inputType-75"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-120 MuiButton-root-94 MuiButton-contained-105 MuiButton-containedPrimary-106 MuiButton-raised-108 MuiButton-raisedPrimary-109"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-95"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-123"
        />
      </button>
    </form>
  </div>
</div>
`;
